src/argilla/client/sdk/v1/datasets/models.py

```diff
@@ -0,0 +1,87 @@
+#  coding=utf-8
+#  Copyright 2021-present, the Recognai S.L. team.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import sys
+from datetime import datetime
+from typing import Any, Dict, List, Optional, Union
+from uuid import UUID
+
+if sys.version_info >= (3, 8):
+    from typing import Literal
+else:
+    from typing_extensions import Literal
```

Comment:
The most robust solution may be to use the Python [EAFP](https://docs.python.org/3/glossary.html#term-EAFP) coding style:
```python
try:
    from typing import Literal
except ImportError:
    from typing_extensions import Literal
```
That way, you'll never accidentally list the wrong `sys` version, and it won't break if in the future the Python devs decide to moving something from `typing` back to `typing_extensions` or something. (I assume they won't, but still).

This extends to the rest of the codebase here. I'm open to your thoughts here.

Reply:
Well since that certainly would work, I guess that if we can catch it better right? Anyway it's a minor issue so I'm also open to catch that on `ImportError` and fallback to `typing_extensions` if `typing` doesn't work for a certain Python version e.g. `typing.Annotated` introduced in Python 3.9, or `typing.Literal` introduced in Python 3.8.